src/metabase/driver/mongo/query_processor.clj

```diff
@@ -211,16 +234,45 @@
 
 ;;; ### filter
 
+;; omf: 'max' date in a date range is next day at 00:00:00.000
+(defn- resolve-between [basetype minvalue maxvalue]
+  (cond (= basetype :type/DateTime)
+    {$gte minvalue
+     $lt (t/plus maxvalue (t/days 1))}
+  :else
```

Comment:
The it would make more sense just to write this like

```clojure
{$gte minvalue
 $lt  (if (isa? basetype :type/DateTime)
        (t/plus maxvalue (t/days 1))
        maxvalue)}
```